### PR TITLE
Add validation to server for user authentication inputs

### PR DESCRIPTION
**Pull request by Tyak99**
### What does this PR do?

* Add email and password validation to user authentication routes

### Type of change
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] Chore (a non-breaking change which doesn't affect existing functionality)

### Description of Task to be completed?

* Add self-explanatory error messages to be sent to the user when wrong inputs are sent along authentication requests 

### How should this be manually tested?

* git clone the repo
* npm install
* npm run dev
* send a POST request to the authentication routes

### What are the relevant pivotal tracker stories?

#164564586

### CODE REVIEW

File: server/test/user.test.js [hunk 1]
```diff
@@ -52,6 +52,57 @@ describe('Test user signup route', () => {
...
+      firstName: 'John',
+      lastName: 'Snow',
+    };
+    chai
+      .request(server)
+      .post('/api/v1/auth/signup')
+      .send(user)
+      .end((err, res) => {
+        expect(res.body.status).to.eql(422);
+        expect(res.body).to.have.property('error');
```
Comment (CEOehis) [inline server/test/user.test.js:68]: add a test to check the error message.

File: server/test/user.test.js [hunk 2]
```diff
@@ -52,6 +52,57 @@ describe('Test user signup route', () => {
...
```
Comment (CEOehis) [inline server/test/user.test.js:86]: add an assertion to check the error message

File: server/test/user.test.js [hunk 3]
```diff
@@ -150,6 +201,21 @@ describe('Test user sign in route', () => {
...
+      email: 'john',
+      password: 'secret',
+    };
+    chai
+      .request(server)
+      .post('/api/v1/auth/login')
+      .send(user)
+      .end((err, res) => {
+        expect(res.body.status).to.eql(422);
+        expect(res.body).to.have.property('error');
```
Comment (CEOehis) [inline server/test/user.test.js:225]: add an assertion to check the error message

**Review by CEOehis — CHANGES_REQUESTED** [review]

File: server/test/user.test.js (hunk 3 above)
Comment (Tyak99) [reply server/test/user.test.js:225]: Done

File: server/test/user.test.js (hunk 2 above)
Comment (Tyak99) [reply server/test/user.test.js:86]: Done

File: server/test/user.test.js (hunk 1 above)
Comment (Tyak99) [reply server/test/user.test.js:68]: Done

**Review by CEOehis — APPROVED** [review]
Looks good

**Pull request merged**